sdk/core/System.ClientModel/src/Internal/SSE/AsyncServerSentEventEnumerator.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Internal;
+
+internal sealed class AsyncServerSentEventEnumerator : IAsyncEnumerator<ServerSentEvent>, IDisposable, IAsyncDisposable
+{
+    // TODO: make this configurable per coming from TypeSpec
+    private static readonly ReadOnlyMemory<char> _doneToken = "[DONE]".AsMemory();
```

Comment:
I think we should not have a default for the terminal event and always as the caller to provide. In fact, maybe the abstraction, should not even have a parameter that we try to match but rather some virtual/abstract API that implementers can implement to exit the enumeration. Of course, our default implementation returned from create could use the parameter and do an exact match (as you do today).

Reply:
Addressed.